features/netsocket/OnboardNetworkStack.h

```diff
@@ -119,6 +119,19 @@ class OnboardNetworkStack : public NetworkStack {
 
         virtual char *get_ip_address(char *buf, nsapi_size_t buflen) = 0;
 
+        /** Copies IPv6 link local address of the network interface to user supplied buffer
+         *
+         * @param    buf        buffer to which IP address will be copied as "W:X:Y:Z"
+         * @param    buflen     size of supplied buffer
+         * @param    interface_name  Network interface name
+         * @return              Pointer to a buffer, or NULL if the buffer is too small
+         */
+
+        virtual nsapi_error_t get_ipv6_link_local_address(SocketAddress *address)
```

Comment:
Why is this here? There is a default implementation in NetworkInterface already for this

Reply:
Some mesh interfaces, for example [LoWPANNDInterface](https://github.com/ARMmbed/mbed-os/blob/419556b84a135c8037bfc162d84074e0ddb38fcc/features/nanostack/mbed-mesh-api/source/LoWPANNDInterface.cpp#L27), don't inherit from `NetworkInterface`. Instead they are required to implement this `Nanostack::Interface` class, which is separate from `NetworkInterface`.
Not sure about the reason behind this duplication, perhaps `NetworkInterface` didn't always match the Nanostack's interface? We somehow don't impose the implementation of `connect` or `disconnect` and I can see `LoWPANNDInterface` implements [its own](https://github.com/ARMmbed/mbed-os/blob/master/features/nanostack/mbed-mesh-api/source/LoWPANNDInterface.cpp#L41) `mesh_connect()`